### PR TITLE
fix(lerobot_local): batch declarative obs when the pipeline has no AddBatchDimension step

### PR DESCRIPTION
**Pull request by cagataycali**
## Problem

The declarative `embodiment=` path in `LerobotLocalPolicy.get_actions` feeds the preprocessed observation straight to the model on the assumption that the preprocessor pipeline always ends in `AddBatchDimension` + `Device` steps that batch every tensor and move it to the policy device:

```python
# no _fixup needed (AddBatchDimension + Device steps in the pipeline handle shape/device)
batch = self._processor_bridge.preprocess(observation, instruction=instruction)
```

That assumption holds for a standard `policy_preprocessor.json`, but **not** for the `norm_stats.json` fallback. When a checkpoint ships no standard pipeline config, the bridge builds a minimal pipeline (`norm_stats.py`):

```python
preprocessor = pipeline_cls(steps=[pre_step_cls(state_norm)])  # normalizer only
```

This pipeline has no batch step and no device step. On the declarative path the model therefore receives an unbatched `observation.state` `(D,)` alongside a batched `(1, C, H, W)` image, producing a `torch.stack` rank mismatch inside `select_action`. The legacy (no-embodiment) branch already guards against this by running `_fixup_preprocessed_batch`; the declarative branch did not.

## Fix

Run the idempotent `_fixup_preprocessed_batch` on the declarative path too. It leaves already-batched, already-CHW, and already-on-device tensors untouched, so it is a **no-op** for standard pipelines (which `AddBatchDimension` already batched) and restores the batched-tensor contract for minimal ones. The batched-tensor contract is now independent of which steps a checkpoint's pipeline happens to ship.

## Reproduction (real lerobot pipeline, network-free)

Driving `get_actions` on the declarative path with a pipeline that lacks `AddBatchDimension` (the exact shape the `norm_stats.json` fallback builds):

```
before: observation.images.front (3, 48, 64),  observation.state (6,)    # rank mismatch
after:  observation.images.front (1, 3, 48, 64), observation.state (1, 6) # batched
```

## Tests

Adds `TestDeclarativeBatchesWithoutAddBatchDimensionStep` in `tests/policies/lerobot_local/test_declarative_single_camera.py`. It builds a real (no-download) `DataProcessorPipeline(steps=[RenameObservationsProcessorStep(...)])` with no batch step, drives the full `get_actions_sync` declarative path, and asserts the model receives `observation.state (1, 6)` and `observation.images.front (1, 3, 48, 64)`. The test fails on pre-fix code (`(6,) != (1, 6)`) and passes after.

## Verification

- `ruff check` / `ruff format --check` clean; `mypy strands_robots tests tests_integ` clean.
- `pytest tests/policies/lerobot_local/` — 459 passed (458 prior + 1 new).
- Docs: added a note to the processor-bridge section of `docs/policies/lerobot-local.md` describing the minimal-fallback pipeline being batched at runtime.

Scope: one branch + one idempotent call; no behavior change on the standard pipeline path.

### CODE REVIEW

**Review by yinsong1986 — APPROVED** [review]
## Summary

This PR fixes a rank-mismatch crash on the declarative `embodiment=` inference path in `LerobotLocalPolicy.get_actions`. The declarative branch previously fed the preprocessed observation straight to the model on the assumption that the pipeline always ends in `AddBatchDimension` + `Device` steps. That holds for a standard `policy_preprocessor.json` but not for the `norm_stats.json` fallback, which builds a normalizer-only `DataProcessorPipeline` with no batch/device step — producing an unbatched `observation.state (D,)` next to a batched `(1, C, H, W)` image and a `torch.stack` rank mismatch inside `select_action`. The fix runs the existing `_fixup_preprocessed_batch` on the declarative path too, matching the legacy branch.

The call is genuinely idempotent: `_fixup_preprocessed_batch` only unsqueezes state tensors when `ndim == 1` and images when `ndim == 3`, so standard-pipeline output (already batched, already CHW, already on device) passes through with at most a defensive device-move / float64→float32 cast — a no-op for the standard path. Scope is one branch plus one idempotent call with no behavior change on the standard pipeline.

## What's good

- Tight scope: a single guarded call mirroring the proven legacy-path guard.
- Idempotency claim verified against the `_fixup_preprocessed_batch` source — no risk of double-batching on the standard pipeline.
- New regression test (`TestDeclarativeBatchesWithoutAddBatchDimensionStep`) builds a real, network-free minimal pipeline and asserts the batched-tensor contract; fails on pre-fix code.
- Comments are plain ASCII, no host paths, docs updated in the same PR.

## Verification suggestions

- `pytest tests/policies/lerobot_local/test_declarative_single_camera.py -k Batches` to confirm the new test passes post-fix and would fail pre-fix.
